### PR TITLE
Bump NN, Maps and common dependency versions to `86.0.2`, `10.3.0-beta.1` and `21.1.0-beta.1` respectively

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

Bumps NN, Maps and common dependency versions to `86.0.2`, `10.3.0-beta.1` and `21.1.0-beta.1` respectively.

cc @SiarheiFedartsou

### CODE REVIEW

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorLoader.kt [hunk 1]
```diff
@@ -86,8 +89,60 @@ internal object NavigatorLoader {
         }
     }
 
+    // TODO Remove after NN disables it by default
     private fun customConfig(deviceProfile: DeviceProfile): String {
```
Comment (Guardiola31337) [inline libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorLoader.kt:93]: When reviewing could you pay special attention to this? cc @SiarheiFedartsou

**Review by cafesilencio — APPROVED** [review]

**Comment by Guardiola31337** [comment]
Noting that currently tests are not compiling.

**Comment by Guardiola31337** [comment]
> Noting that currently tests are not compiling.

Fixed.

**Review by abhishek1508 — APPROVED** [review]

**Review by kmadsen — APPROVED** [review]

**Comment by kmadsen** [comment]
Same instrumentation failure is happening on this PR https://github.com/mapbox/mapbox-navigation-android/pull/5103

https://github.com/mapbox/mapbox-navigation-android/blob/d5823fd49f5f0559004b4340c95b86683aca78ec/instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/navigationview/NavigationViewExternalObserversTests.kt#L89-L105

```
org.junit.ComparisonFailure: expected:<[Head east on Fulton Street, then turn left onto 27th Avenue]> but was:<[Drive east on Fulton Street. Then, in a quarter mile, Turn left onto 27th Avenue.]>
at org.junit.Assert.assertEquals(Assert.java:117)
at org.junit.Assert.assertEquals(Assert.java:146)
```

The instruction has the same meaning, but different words.

**Comment by kmadsen** [comment]
considering that test is failing on multiple places and we're looking to get an alpha out, think this can be merged. 

unless there is another obvious issue 🤔

**Comment by kmadsen** [comment]
Testing manually and voice instructions are working. Disabled the requirement to pass instrumentation-tests because it does not seem like a blocker for `2.3.0-alpha.1`

**Pull request merged**